blockchain/chain_sync/src/network_context.rs

```diff
@@ -70,6 +70,27 @@ impl SyncNetworkContext {
             .map(|fts| fts.tipset().map_err(|e| e.to_string()))
             .collect::<Result<_, _>>()
     }
+    /// Send a blocksync request for full tipsets (includes messages)
+    pub async fn blocksync_fts(
+        &mut self,
+        peer_id: PeerId,
+        tsk: &TipSetKeys,
+        count: u64,
+    ) -> Result<FullTipset, String> {
+        let bs_res = self
+            .blocksync_request(
+                peer_id,
+                BlockSyncRequest {
+                    start: tsk.cids().to_vec(),
+                    request_len: count,
+                    options: 1,
```

Comment:
This doesn't request messages as is, so I don't see the point of this. Also would be preferable to have just one function with this logic and have the headers and fts functions just call with the options (remove duplicate logic)
```suggestion
                    options: 3,
```